File: t_test/t_test.R
# t-tests

one_sample_t_test <- function(x,y) {
  # one sample t test
  # x = vector
  # y = single value
  # returns t_value
  mean_x <- sum(x)/length(x)
  variance_x <- sum((x - mean_x)^2)/(length(x)-1)
  t_value <- (mean_x - y) / (sqrt(variance_x) / sqrt(length(x)))
  t_value
}

File: t_test/t_test.py
# t tests

def one_sample_t_test(x, y):
    """one sample t test
    x = list
    y = single value
    return = t_value as float"""
    mean_x = sum(x)/float(len(x))
    print(mean_x)
    variance_x = sum(([(i - mean_x)**2 for i in x]))/float(len(x) - 1)
    print(variance_x)
    t_value = (mean_x - y) / float((variance_x**(0.5)/ len(x)**(0.5)))
    return t_value